staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -263,10 +263,12 @@ type Cacher struct {
 	// indexedTrigger is used for optimizing amount of watchers that needs to process
 	// an incoming event.
 	indexedTrigger *indexedTriggerFunc
-	// watchers is mapping from the value of trigger function that a
-	// watcher is interested into the watchers
+	
 	watcherIdx int
-	watchers   indexedWatchers
+	// watchers is mapping from both watcherIdx and value of trigger function that a watcher is interested in to the watchers,
+	// when triggerSupported is true by the indexedTrigger. 
+	// If triggerSupported is false, mapping exists between watcherIdx and the watcher that the index identifies.
+	watchers indexedWatchers
```

Comment:
nit: this is kinda difficult to read, could you try simplifying things a bit here?

Reply:
The value of trigger function can be either True or False. So, maybe removing those lines can do good.

Reply:
The value of `triggerSupported` seems to be an implied statement from that watchers is gonna map, i.e if True, any event that interests the watcher is mapped & if False, it maps to the watcher that the index identifies. 
So, maybe removing that line can do good.